generator/feature_segments_checker/feature_segments_checker.cpp

```diff
@@ -75,14 +79,37 @@ void WriteCSV(Cont const & cont, string const & fileName)
     fout << a.first << "," << a.second << endl;
 }
 
-/// \returns expected point altitude in meters according to linear model.
-double ExpectedPointAltitude(int16_t startAltitudeMeters, int16_t endAltitudeMeters,
-                             double distFromStartMeters, double featureLengthMeters)
+/// \returns y = k * x + b. It's an expected altitude in meters.
+double GetY(double k, double b, double d)
 {
-  if (featureLengthMeters == 0.0)
-    return 0;
-  double const k = (endAltitudeMeters - startAltitudeMeters) / featureLengthMeters;
-  return startAltitudeMeters + k * distFromStartMeters;
+  return k * d + b;
+}
+
+/// \brief Calculates factors |k| and |b| of a line using linear least squares method.
+/// \returns false in case of error (e.g. if the line is parallel with ordinate axis)
+/// and true otherwise.
+bool LinearLeastSquaresFactors(vector<double> const & xs, vector<double> const & ys, double & k, double & b)
+{
+  size_t const n = xs.size();
+  long double mx = 0, my = 0, mxy = 0, mx2 = 0;
```

Comment:
Sorry, but I don't see any benefits of long double over double. sizeof(long
double) is unspecified and varies on different platforms, therefore results
computed with long double are not representative and reproducible.
Moreover, long double are usually slower than doubles, because of some
strange sizes, like 10 or 12 bytes. So, please, don't use long double just
because it's wider (sometimes).

On 20 June 2016 at 11:43, Vladimir Byko-Ianko <notifications@github.com>
wrote:

> In generator/feature_segments_checker/feature_segments_checker.cpp
> <https://github.com/mapsme/omim/pull/3503#discussion_r67653852>:
>
> >  {
> > -  if (featureLengthMeters == 0.0)
> > -    return 0;
> > -  double const k = (endAltitudeMeters - startAltitudeMeters) / featureLengthMeters;
> > -  return startAltitudeMeters + k * distFromStartMeters;
> > +  return k * d + b;
> > +}
> > +
> > +/// \brief Calculates factors |k| and |b| of a line using linear least squares method.
> > +/// \returns false in case of error (e.g. if the line is parallel with ordinate axis)
> > +/// and true otherwise.
> > +bool LinearLeastSquaresFactors(vector<double> const & xs, vector<double> const & ys, double & k, double & b)
> > +{
> > +  size_t const n = xs.size();
> > +  long double mx = 0, my = 0, mxy = 0, mx2 = 0;
>
> I don't thing it's an important question. I guess double and long double
> suits for these case. And I asked you about advantage of long double.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mapsme/omim/pull/3503/files/0ec6e0a5a8248cca254e44bdc2da2a7017052e4b#r67653852>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAbvp5dLAfBUlmj1K8kjHPqMyMp0ZhoBks5qNlKngaJpZM4Ix8Rp>
> .
>